lib/Publisher.js

```diff
@@ -140,12 +147,18 @@ class Publisher extends EventEmitter {
    * Pulls all msgs off queue, serializes, and publishes them
    */
   _handleMsgQueue(msgQueue) {
+
+    // There's a small chance that we were shutdown while the spinner was locked
+    // which could cause _handleMsgQueue to be called if this publisher was in there.
+    if (this.isShutdown()) {
```

Comment:
This looks really familiar. Roscom deals with a very similar edge case, but in the most complex way possible with lots of smart pointers. I'm glad you don't have to deal with that kind of nonsense here.